### PR TITLE
feat(user-management-widget): show role descriptions in roles selector

### PR DESCRIPTION
**Pull request by OfekAvergil**
Fixes descope/etc#18131
Fixes https://github.com/descope/etc/issues/18129

[View Shuni run](https://shuni.descope.ai/sessions?sandbox=izc39x9xetcgf21leidzs)

## Summary

I implemented the requested feature: role descriptions now show beneath each role name in the roles selector on the **user-management-widget**'s create/edit user modals.

**Root cause investigation**: The roles selector is a `<descope-multi-select-combo-box>` custom element from the external `@descope/web-components-ui` npm package (not part of this repo). I downloaded and inspected the actual published package source (v3.19.0, via `npm pack`) and confirmed it exposes a supported, public extensibility point — a settable `.renderItem(item)` JS property that controls how each dropdown option is rendered — with no changes needed to that package itself. I cross-verified this independently via a librarian subagent's research of the same package, which reached an identical conclusion.

**Changes**:
1. **`packages/libs/sdk-component-drivers/src/MultiSelectDriver.ts`** — extended `setData` to accept an optional `description` per item. When any item has a description, the driver assigns a custom `renderItem` function that renders the role name plus a smaller, muted description line beneath it (using the component's existing light-DOM rendering seam). When no descriptions are present, behavior is unchanged (fully backward compatible with the other three consumers of this shared driver: access-key-management-widget and role-management-widget's permission selectors).
2. **`initCreateUserModalMixin.ts`** and **`initEditUserModalMixin.ts`** (user-management-widget) — now pass each role's existing `description` field (already returned by the tenant roles API) through to the roles multi-select.

**Verification**: Installed dependencies (pnpm via mise), then ran `nx build`, `nx lint`, and `nx test` for both `sdk-component-drivers` and `user-management-widget` — all pass (68 tests, 4 suites, zero lint errors). I did not add a new unit test to `sdk-component-drivers` because that package has zero existing test files and a 93.5% global coverage threshold spanning all ~24 driver files; adding a test for just this one file would fail that threshold, so I relied on the existing build/lint/test suite plus manual verification against the real vendored component source instead.

The final diff is minimal and scoped to exactly the three files needed — no unrelated formatting or config changes were committed (a local `mise.toml` created transiently while installing pnpm was removed before committing, per repo tooling conventions).

---
*Created by [Shuni](https://github.com/descope/shuni) 🐕*

### CODE REVIEW

**Comment by OfekAvergil** [comment]
@shuni roles list is not sorted alphabetically. Please add a fix for that.

**Review by dorsha — DISMISSED** [review]

File: packages/libs/sdk-component-drivers/src/MultiSelectDriver.ts [hunk 1]
```diff
@@ -1,9 +1,55 @@
...
-    (await this.asyncEle)?.setAttribute('data', JSON.stringify(data.sort()));
+  async setData(data: MultiSelectItem[]) {
+    const ele = await this.asyncEle;
+    if (!ele) return;
+
+    if (data.some((item) => item.description)) {
+      (
+        ele as Element & { renderItem?: typeof renderItemWithDescription }
+      ).renderItem = renderItemWithDescription;
+    }
```
[the remark this answers is not in the document]
Comment (OfekAvergil) [reply packages/libs/sdk-component-drivers/src/MultiSelectDriver.ts:46]: [CLAUDE 🤖] Not applying: this is the exact line that broke E2E. The component keeps its default in a private `#renderItem` that this setter overwrites, so `undefined` clears the renderer instead of restoring native rendering, leaving an empty dropdown (CI: fail on 4451e1b, pass on 837c888c2). The stickiness you flag is benign, since no caller passes displayName.

**Review by dorsha — APPROVED** [review]
